src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,53 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private T[] elements;
+    private int size;
+    private static final int INITIAL_CAPACITY = 10;
+
+    @SuppressWarnings("unchecked")
+    public MateStack() {
+        elements = (T[]) new Object[INITIAL_CAPACITY];
+        size = 0;
+    }
 
+    public void push(T value) {
+        if (size == elements.length) {
+            resize();
+        }
+        elements[size++] = value;
     }
 
     public T peek() {
-        return null;
+        throwExeption();
+        return elements[size - 1];
     }
```

Comment:
Bug: The method 'throwExeption' should be called after checking that the stack is not empty. In the current implementation, an exception will be thrown every time the method is called, even if the stack is not empty.